src/Serialization.js

```diff
@@ -173,8 +190,15 @@ exports.setTxBodyMint = setter("mint");
 exports.newMint = () => lib.Mint.new();
 
 exports._bigIntToInt = maybe => bigInt => {
+  // this is needed because try/catch overuse breaks runtime badly
+  // https://github.com/Plutonomicon/cardano-transaction-lib/issues/875
   try {
     const str = bigInt.to_str();
+    check_limit(
+      BigInt(str),
+      BigInt("-18446744073709551615"),
+      BigInt("18446744073709551616")
+    ); // 2 ^ 64
```

Comment:
```suggestion
    check_limit(
      bigInt,
      BigInt("-18446744073709551615"),
      BigInt("18446744073709551616")
    ); // 2 ^ 64
    const str = bigInt.to_str();
```

Reply:
@jy14898 what did you mean by this suggestion? The first BigInt is CSL BigInt, and the other is native. Can't pass one instead of the other.